### PR TITLE
Fixed #30497 -- Ignored document type in assertXMLEqual()/assertXMLNotEqual().

### PR DESCRIPTION
**Pull request by caioariede**
Trac Ticket: https://code.djangoproject.com/ticket/30497

### CODE REVIEW

File: tests/test_utils/tests.py [hunk 1]
```diff
@@ -920,6 +920,19 @@ def test_simple_not_equal_with_whitespace_in_the_middle(self):
         xml2 = "<elem>foo</elem> <elem>bar</elem>"
         self.assertXMLNotEqual(xml1, xml2)
 
+    def test_doctype_declaration_equal(self):
```
Comment (felixxm) [inline tests/test_utils/tests.py]: `test_doctype_declaration_equal` -> `test_doctype_root`

File: tests/test_utils/tests.py [hunk 2]
```diff
@@ -920,6 +920,19 @@ def test_simple_not_equal_with_whitespace_in_the_middle(self):
...
+        xml1 = '''
+            <?xml version="1.0" encoding="UTF-8"?>
+            <!DOCTYPE root SYSTEM "example.dtd">
+            <root />
+        '''
+        xml2 = '''
+            <?xml version="1.0" encoding="UTF-8"?>
+            <!DOCTYPE root SYSTEM "example.dtd">
+            <root />
+        '''
```
Comment (felixxm) [inline tests/test_utils/tests.py]: I think we can use one line examples, e.g.
```python
        xml1 = '<?xml version="1.0"?><!DOCTYPE root SYSTEM "example.dtd"><root />'
        xml2 = '<?xml version="1.0"?><!DOCTYPE root SYSTEM "example.dtd"><root />'
```

**Comment by claudep** [comment]
If there's a possible comparison of `Node.DOCUMENT_TYPE_NODE`s, I think we should do it.
@caioariede Could you explore that please?

**Comment by felixxm** [comment]
I agree with Claude.

**Comment by caioariede** [comment]
@felixxm @claudep I thought about `Node.DOCUMENT_TYPE_NODE` comparisons and tested the behavior of `xmldiff` library and it does not compare `Node.DOCUMENT_TYPE_NODE`.

Also, we do not perform any comparisons on xml declaration, for example:

```python
self.assertXMLEqual(
    '<?xml version="1.0" encoding="UTF-8"?><root />',
    '<?xml version="2.0" encoding="UTF-8"?><root />')
```

So I'm not really sure if we should test these things, as this is not a string comparison utility but a XML comparison utility, which I guess should be more concerned about root element and what is inside it.

File: tests/test_utils/tests.py (hunk 2 above)
Comment (caioariede) [reply tests/test_utils/tests.py]: TBH, I think multiple lines make it more readable. If that's just a suggestion ¯\_(ツ)_/¯

**Comment by claudep** [comment]
Makes sense, thanks. Is it worth documenting that in the docs?

**Comment by caioariede** [comment]
@claudep yeah, that I think it's worth mentioning it in the docs 👍

**Comment by felixxm** [comment]
@caioariede Thanks!

**Pull request merged**